README.md

```diff
@@ -64,13 +64,15 @@ The memory counter uses `collectgarbage("count")` to get the amount of Lua memor
 3. Using mem.lua - Draw memory usage at specified position or get current memory usage
 
 
-### mem.create(format, position, color)
+### mem.create(format, position, color, show_avarage, avarage_interval)
```

Comment:
I'm not a big fan of functions with many optional parameters. Having format, position and color all being optional already it becomes unwieldy to add even more. I would suggest one of:

```
-- all of these are optional and have a default value if not specified
local options = {
    format = "MEM %dkb",
    position = vmath.vector3(100, 100, 0),
    color = vmath.vector4(1, 0, 0, 1),
    show_average = true,
    avarage_interval = 60,
}
mem.create(options)
```

Or keep format, position and color as arguments and the new ones as options:

```
local options = {
    show_average = true,
    avarage_interval = 60,
}
mem.create(format, position, color, options)
```

Final thought, what if we only have one argument which is the interval? And it defaults to 1 to give you the default behavior of showing the memory for each frame instead of an average.

```
local options = {
    average = 60, -- calculate average memory usage over 60 samples
}
mem.create(format, position, color, options)
```

One more thing: An alternative approach, or a different option, would be to only update the text  at some interval and show the value at that specific frame instead of the average. It's not the same and I'm not sure if it is better or worse than showing an average?

PS It's spelled average with an e, not avarage!

Reply:
I don't like multiple options either, just tried to not break anything else.

But of the two options, i prefer the mem.create(options) the best, could of course also be written without a variable:

```
mem.create({
    format = "MEM %dkb",
    position = vmath.vector3(100, 100, 0),
    color = vmath.vector4(1, 0, 0, 1),
    show_average = true,
    avarage_interval = 60,
})
```

I liked your idea about having a interval with default to 1.

In our project, the memory usage fluctuates between 300-600 currently, so if we don't show an average (thanks for the spelling), it will jump a lot between 300 and 600. So if I do some optimizations, it's hard to see any differences if I can't see an average over time. But maybe looking at Lua memory usagae as a total is not the best when optimizing. But it gives a rough number at least.

If we go with that, then it's "just" one more optional parameter, so it's a non breaking change in the API. Ex:

```
mem.create(format, position, color, interval)
```

What do you think about that? @britzl

Reply:
Ok, let's agree on `mem.create(format, position, color, interval)` this time and next time we want to add something we switch to an `options` table instead!